modules/apps/transfer/internal/convert/convert.go

```diff
@@ -0,0 +1,54 @@
+package convert
+
+import (
+	"strings"
+
+	"github.com/cosmos/ibc-go/v8/modules/apps/transfer/types"
+)
+
+// PacketDataV1ToV2 converts a v1 packet data to a v2 packet data.
+func PacketDataV1ToV2(packetData types.FungibleTokenPacketData) types.FungibleTokenPacketDataV2 {
+	if err := packetData.ValidateBasic(); err != nil {
+		panic(err)
+	}
+
+	v2Denom, trace := ExtractDenomAndTraceFromV1Denom(packetData.Denom)
+	return types.FungibleTokenPacketDataV2{
+		Tokens: []types.Token{
+			{
+				Denom:  v2Denom,
+				Amount: packetData.Amount,
+				Trace:  trace,
+			},
+		},
+		Sender:   packetData.Sender,
+		Receiver: packetData.Receiver,
+		Memo:     packetData.Memo,
+	}
+}
+
+// extractDenomAndTraceFromV1Denom extracts the base denom and remaining trace from a v1 IBC denom.
```

Comment:
```suggestion
// ExtractDenomAndTraceFromV1Denom extracts the base denom and remaining trace from a v1 IBC denom.
```
looks like it wasn't made private?